workflow-templates/gitleaks.yml

```diff
@@ -1,18 +1,6 @@
 name: GitLeaks Secret Scan
 
-# Only scans the main branch by default.
-# It's recommended to add other branches
-# to catch commited secrets sooner.
-on:
-  push:
-    branches:
-      - $default-branch
-  pull_request:
-    branches:
-      - $default-branch
-  workflow_dispatch:
-    branches:
-      - $default-branch
+on: [push, pull_request, workflow_dispatch]
 
```

Comment:
Unless this is a _really_ fast scan, I would strongly suggest that we don't do this on everything. In all honesty, this should be a pre-commit hook so it never even gets committed with a secret, but I'm wary of this being problematically slow if we do this on every push or pull request to every branch.